### PR TITLE
MAXMIS-745/859 Add cypress based test suite for the Civil operational map application

### PR DESCRIPTION
**Pull request by MaeveOReilly**
## What

Add a test suite which runs a cypress test we run in travis (against mocked data) in FVT (against real data).  The intention is only to add the type of UI smoke tests we do before release.  The bulk of the UI testing will stay in travis only. We expect/hope that using same test code/framework in FVT as we have in Travis will reduce our cost of maintenance, and will break at build time before it can get as far deploy (FVT) time. 

## Issues
Epic - https://jsw.ibm.com/browse/MASR-1891

## Testing 

Tested locally and in personal fvt - https://dashboard.masdev.wiotp.sl.hursley.ibm.com/tests/fvtstable/testsuite/ibm-mas-manage/civil-api-defectdetection

image

### CODE REVIEW

**Comment by unnati-solanki-git** [comment]
@MaeveOReilly Can you share the pipeline sequence to visualize how it looks like!

File: tekton/src/tasks/fvt/fvt-manage-cypress.yml.j2 [hunk 1]
```diff
@@ -0,0 +1,204 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: mas-fvt-manage-cypress
```
Comment (bmoferrall) [inline tekton/src/tasks/fvt/fvt-manage-cypress.yml.j2]: I can't find where this task is referenced anywhere Maeve?
(Just trying to understand the flow for my sins)

**Comment by MaeveOReilly** [comment]
> @MaeveOReilly Can you share the pipeline sequence to visualize how it looks like!

Hey Unnati - do you mean with 'pipeline sequence' a screenshot from openshift of it running? It comes directly after the manage-setup-is.  I'll have to rerun pfvt to get that. 

Edit - Rerunning [pfvt](https://v3.travis.ibm.com/github/maximoappsuite/fvt-personal/builds/29416352) to get screenshot and removed unused task (thanks @bmoferrall ).

**Comment by MaeveOReilly** [comment]
@unnati-solanki-git if there a way to make screenshots with the whole pipeline it isn't obvious to me! 

left
image

right
image

So the fvt-ibm-mas-manage-cypress comes after the fvt-manage-setup-is, which is the only dependency (that is where the user + map is configured).

File: tekton/src/tasks/fvt/fvt-manage-cypress.yml.j2 (hunk 1 above)
Comment (MaeveOReilly) [reply tekton/src/tasks/fvt/fvt-manage-cypress.yml.j2]: Tested, is fine without it.  Thanks.

**Comment by unnati-solanki-git** [comment]
> > @MaeveOReilly Can you share the pipeline sequence to visualize how it looks like!
> 
> Hey Unnati - do you mean with 'pipeline sequence' a screenshot from openshift of it running? It comes directly after the manage-setup-is. I'll have to rerun pfvt to get that.
> 
> Edit - Rerunning [pfvt](https://v3.travis.ibm.com/github/maximoappsuite/fvt-personal/builds/29416352) to get screenshot and removed unused task (thanks @bmoferrall ).

There is quick and easy way to do this. I learnt it from @alequint 

Here is the how: https://github.com/ibm-mas/cli/pull/1131#issuecomment-2243754022

Hope this helps in future PRs :)

**Comment by unnati-solanki-git** [comment]
@MaeveOReilly This [sequence](https://github.com/ibm-mas/cli/pull/1317#issuecomment-2471514897) looks great 

Just one thing now we have to do after this got merged. We need to observe that how it behaves in the official pipeline for next couple of runs considering the fact that it runs in parallel to selenium tests (fvt-manage-maintmanager-setup) 👍 

Thanks!

**Review by unnati-solanki-git — APPROVED** [review]

**Comment by MaeveOReilly** [comment]
I had run it locally against the civil cluster as well
[local.txt](https://github.com/user-attachments/files/17728326/local.txt)

**Review by bmoferrall — APPROVED** [review]

**Review by whitfiea — APPROVED** [review]

**Pull request merged**